### PR TITLE
Add registry mirrors support in kvm

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/12064

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/giantswarm/certs v0.2.0
 	github.com/giantswarm/errors v0.2.3
 	github.com/giantswarm/k8sclient v0.2.0
-	github.com/giantswarm/k8scloudconfig/v6 v6.4.0
+	github.com/giantswarm/k8scloudconfig/v7 v7.0.0
```
Comment (tfussell) [inline go.mod:10]: Might be good to use v7.0.3 here to get rid of the missed v6 import (https://github.com/giantswarm/k8scloudconfig/blob/master/CHANGELOG.md#701---2020-07-20).

File: service/controller/cluster.go [hunk 2]
```diff
@@ -44,6 +45,8 @@ type ClusterConfigOIDC struct {
 	GroupsPrefix   string
 }
 
+type Registry struct{}
```
Comment (tfussell) [inline service/controller/cluster.go:48]: Is this used?

Comment (corest) [reply service/controller/cluster.go:48]: good catch

File: service/controller/cloudconfig/cloud_config.go [hunk 3]
```diff
@@ -134,6 +136,7 @@ func New(config Config) (*CloudConfig, error) {
 
 		ignitionPath:    config.IgnitionPath,
 		k8sAPIExtraArgs: k8sAPIExtraArgs,
+		registryMirrors: config.RegistryMirrors,
```
[the remark this answers is not in the document]
Comment (corest) [reply service/controller/cloudconfig/cloud_config.go:139]: `RegistryDomain` is not used in cloudconfig resource. Only in `configmap`

File: service/controller/cloudconfig/master_template.go [hunk 4]
```diff
@@ -35,6 +35,7 @@ func (c *CloudConfig) NewMasterTemplate(customObject v1alpha1.KVMConfig, data Ig
 		params.Kubernetes.Apiserver.CommandExtraArgs = c.k8sAPIExtraArgs
 		params.Images = data.Images
```
[the remark this answers is not in the document]
Comment (corest) [reply service/controller/cloudconfig/master_template.go:36]: Those images are configured with registry in `configmap` resource.
https://github.com/giantswarm/kvm-operator/blob/master/service/controller/resource/configmap/desired.go#L69

File: service/controller/cloudconfig/cloud_config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (kopiczko) [reply service/controller/cloudconfig/cloud_config.go:139]: We should set both `params.RegistryDomain` and `params. RegistryMirrors` in `NewMasterTemplate` and `NewWorkerTemplate`.

[the remark this answers is not in the document]
Comment (corest) [reply service/controller/cloudconfig/cloud_config.go:139]: We pass domain in configmap resource

[the remark this answers is not in the document]
Comment (kopiczko) [reply service/controller/cloudconfig/cloud_config.go:139]: Okay I'm asking to set a param I removed myself in https://github.com/giantswarm/k8scloudconfig/pull/761 🤦

**Pull request merged**